### PR TITLE
[ENG-5102] Add save and publish buttons with validation

### PR DESCRIPTION
**Pull request by bp-cos**
-   Ticket: [ENG-5102]
-   Feature flag: n/a

## Purpose

Add save and publish buttons with validation

## Summary of Changes

Updated the cedar editor, added listeners to valid the metadata record is valid.

## Screenshot(s)

Screenshot 2024-01-31 at 12 42 39 PM

Screenshot 2024-01-31 at 12 42 54 PM

## Side Effects

Nothing

## QA Notes

Test!

[ENG-5102]: https://openscience.atlassian.net/browse/ENG-5102?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/osf-components/addon/components/cedar-metadata-editor/component.ts [hunk 1]
```diff
@@ -63,9 +82,11 @@ export default class CedarMetadataEditor extends Component<Args> {
             record.target = this.args.target;
         }
 
+        record.isPublished = this.isValid;
```
Comment (futa-ikeda) [inline lib/osf-components/addon/components/cedar-metadata-editor/component.ts:85]: Is this `isPublished` flag something we want to be setting on the FE? This seems like a field that should be updated on the BE

File: lib/osf-components/addon/components/cedar-metadata-renderer/component.ts [hunk 2]
```diff
@@ -18,6 +19,15 @@ export default class CedarMetadataRenderer extends Component<Args> {
     viewerConfig = config.cedarConfig.viewerConfig;
     @tracked isShowEditor = false;
 
+    public get hasWritePermission(): boolean {
+        const node = this.args.cedarMetadataRecord.target as NodeModel;
+        if (node.get('modelName') === 'file') {
+            return (node as any).get('target').get('userHasWritePermission');
+        } else {
+            return node.get('userHasWritePermission');
+        }
```
Comment (futa-ikeda) [inline lib/osf-components/addon/components/cedar-metadata-renderer/component.ts:28]: I think the names/types here are a bit confusing/misleading. It's a little bit more verbose, but I think these names and types are a bit more accurate/clear
```suggestion
        const target = this.args.cedarMetadataRecord.target as NodeModel | FileModel;
        if (target.get('modelName') === 'file') {
            return (target as FileModel).get('target').get('userHasWritePermission');
        } else {
            return (target as NodeModel).get('userHasWritePermission');
        }
```

File: lib/osf-components/addon/components/metadata/metadata-detail/component.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+export default class MetadataDetails extends Component<DetailArgs> {
+    @service media!: Media;
+
+    public get hasWritePermission(): boolean {
+        const node = this.args.target as NodeModel;
+        if (node.get('modelName') === 'file') {
+            return (node as any).get('target').get('userHasWritePermission');
+        } else {
+            return node.get('userHasWritePermission');
+        }
```
Comment (futa-ikeda) [inline lib/osf-components/addon/components/metadata/metadata-detail/component.ts:21]: Ditto on the type/variable names here

File: mirage/scenarios/cedar-metatdata-record.ts [hunk 4]
```diff
@@ -16,16 +17,19 @@ export function cedarMetadataRecordsScenario(
...
 
 function threeCedarMetadataRecords(server: Server) {
     const newNode = server.create('node', {
         id: '3-cedar-records',
+        currentUserPermissions: [Permission.Read],
     });
 
     const cedarMetadataRecords = server.createList('cedar-metadata-record', 2);
-    cedarMetadataRecords.push(server.create('cedar-metadata-record', 'isDraft'));
+    // cedarMetadataRecords.push(server.create('cedar-metadata-record', 'isDraft'));
```
Comment (futa-ikeda) [inline mirage/scenarios/cedar-metatdata-record.ts:31]: Can remove this commented out line

**Review by futa-ikeda — CHANGES_REQUESTED** [review]
Just a few minor things I noticed. Good stuff!

**Review by adlius — CHANGES_REQUESTED** [review]

**Pull request merged**